cloud/blockstore/libs/diagnostics/profile_log_ut.cpp

```diff
@@ -388,6 +392,27 @@ Y_UNIT_TEST_SUITE(TProfileLogTest)
             env.EventProcessor.FlatMessages[0]
         );
     }
+
+    Y_UNIT_TEST(TestFlushOnDestruct)
+    {
+        TEnv env;
+        env.ProfileLog->Write(
+            {"disk2",
+             TInstant::Seconds(3),
+             IProfileLog::TReadWriteRequest{
+                 EBlockStoreRequest::WriteBlocks,
+                 TDuration::MilliSeconds(300),
+                 TDuration::MilliSeconds(42),
+                 TBlockRange64::WithLength(10, 20),
+             }});
+        env.Scheduler = nullptr;
+        env.ProfileLog = nullptr;
```

Comment:
you can `ProfileLog.reset(CreateProfileLogStub())` then you don't need to add everywhere the 
```c++
if (ProfileLog) {
}
```